### ISSUE TITLE
Implement an example context and hooks. 

### ISSUE THREAD

**Issue by dopry** [ory/kratos-nextjs-react-example#12]
While the example is useful for getting started with Ory out of the box, it's isn't a very friendly approach for large sites. Most react authentication libraries use a combination of Providers/Context to store the client to the authentication library and hooks to share the data to components.  

https://github.com/authts/react-oidc-context provides an example of a more typical react authentication provider. 

The advantages to this you only initialize the SDK once and share it across you entire app, without needing a useEffect on every page. 
based on my remedial understanding of the libraries, the following would be nice to have. 

**components**
- `<OryIdentityProvider OrySdkUrl="">`  - initializes SDK
- `<RequireMinAAL aal="aal2" onAALInsufficient={() => { console.log("We should probably prompt the user to elevate")}>` - checks users AAL, prevents rendering of child elements and instead allow developers to automate some user defined action when AAL is insufficient maybe showing an elevation dialog 

**hooks**
- `useSession` - get the current session
- `useIdentity` - get the current identity
- `useLogoutUrl` - get the logout url

**Comment by aeneasr** [comment]
I totally agree with you @dopry! We wanted to keep this example as unopinionated and raw as possible, so that users understand how the underlying things work. We have it on our backlog to create high-order components and hooks /fyi @mszekiel @Benehiko

**Comment by dopry** [comment]
@aeneasr glad to hear it! If I get time I may do some of that work as I'm exploring next@13 and Ory on a side project. 

TBH, when I'm wearing my front end dev hat, I don't want to think to much about how Auth works under the hood. The just kind of want to setup the Provider once and know how to check permissions, signup, login, logout. I'm more concerned about stuff like styling components and mapping data around my app. Auth is kind of a secondary concern in my development.  I'm getting it from a vendor like Okta, Auth0, Ory, etc I'm expecting it to just work and I'm going to use the API you give me. I don't want to get under the hood.

Wearing my architect/full stack/security hat, I do want to dive under the hood. I need to audit it. The code sample didn't help me understand how underlying things work. I felt like I was copy pasting code without really understanding what it did. I would have really liked to have better documentation for the sdk listing the methods and types, explaining their arguments, and what they do to the app

Take `createApiHandler` for instance. What endpoints does it create? What does each endpoint do? What arguments do they take?  I had to dig in the source to figure out /ui/login could take a return_to query param to redirect users to a different page post login based on where the user logged in from. A link to https://github.com/ory/integrations/blob/main/src/next-edge/index.ts in the code would start to be helpful, but even then I only  clearly see the Options. How it works isn't very clear.  1) it just maps allsearch params from the query. I have not idea which once are accepted.  I see a ui/welcome redirect that goes to some hosted ui/welcome, but I don't know where to go see that or find out more about it.  Then I see it proxies to the baseURL.  I'm assuming from the read, that is a proxy thar forwards reqeuts to the ory network, but that is an assumption.  Why exactly do I need this proxy? 

Moving on to index.tsx, It would be nice to have  a link to docs for V0alpha2Api and it's configuration structure. It would be nice ice V0Alpha2Api were called something more meaningful like KratosClientApi.  In lieu of sdk docs it would be nice to have links to each method like https://github.com/ory/sdk/blob/2324e5c13c45d43c7e7b3cdc5ebc86b86b627867/clients/kratos/typescript/api.ts#L5475.  but even then the JS doc is in super long lines that is hard to read on GH. 


-

**Comment by OliverGeneser** [comment]
A Providers/Context/Hooks example would really be great!
